src/modules/flow/netctl/netctl.c

```diff
@@ -0,0 +1,336 @@
+
+    name = sol_netctl_service_get_name(service);
+
+    if (sol_netctl_service_get_state(service) ==
+        SOL_NETCTL_SERVICE_STATE_REMOVE) {
+        r = sol_flow_send_string_packet(node,
+            SOL_FLOW_NODE_TYPE_NETCTL_LIST_SERVICES__OUT__REMOVED,
+            name);
+        if (r < 0)
+            SOL_WRN("Failed to send removed service name: %s", name);
+        return;
+    }
+
+    r = sol_flow_send_string_packet(node,
+        SOL_FLOW_NODE_TYPE_NETCTL_LIST_SERVICES__OUT__ADDED,
+        name);
+    if (r < 0)
+        SOL_WRN("Failed to send added service name: %s", name);
+}
+
+static int
+open_list_services(struct sol_flow_node *node, void *data, const struct sol_flow_node_options *options)
+{
+    int r;
+
+    r = sol_netctl_add_service_monitor(service_list_cb, node);
+    SOL_INT_CHECK(r, < 0, r);
+
+    return 0;
+}
+
+static void
+close_list_services(struct sol_flow_node *node, void *data)
+{
+    static const char *errmsg = "Could not delete service monitor";
+
+    if (sol_netctl_del_service_monitor(service_list_cb, node))
+        SOL_WRN("%s", errmsg);
+}
+
+void
+service_status_cb(void *data, const struct sol_netctl_service *service)
+{
+    struct sol_flow_node *node = data;
+    struct network_service_data *mdata = sol_flow_node_get_private_data(node);
+    enum sol_netctl_service_state current_state;
+    const char *name;
+    int r;
+
+    name = sol_netctl_service_get_name(service);
+
+    if (!name || strcmp(name, mdata->service_name)) 
+        return;
+
+    current_state = sol_netctl_service_get_state(service);
+    if(current_state != mdata->state) {
+        r = sol_flow_send_string_packet(node,
+                    SOL_FLOW_NODE_TYPE_NETCTL_SERVICE__OUT__OUT,
+                    sol_netctl_service_state_to_str(current_state));
+        SOL_INT_CHECK_GOTO(r, < 0, error);
+
+        if (mdata->state == SOL_NETCTL_SERVICE_STATE_ONLINE) {
+             r = sol_flow_send_bool_packet(node,
+                         SOL_FLOW_NODE_TYPE_NETCTL_SERVICE__OUT__ONLINE,
+                         false);
+             SOL_INT_CHECK_GOTO(r, < 0, error);
+        }
+
+        if (current_state == SOL_NETCTL_SERVICE_STATE_ONLINE) {
+              r = sol_flow_send_bool_packet(node,
+                         SOL_FLOW_NODE_TYPE_NETCTL_SERVICE__OUT__ONLINE,
+                         true);
+             SOL_INT_CHECK_GOTO(r, < 0, error);
+        }
+
+        mdata->state = current_state;
+
+        return;
+    }
+
+error:
+    mdata->state = current_state;
+
+    SOL_WRN("service %s failed to send online state", name);
+}
+
+void
+error_cb(void *data, const struct sol_netctl_service *service,
+    unsigned int error)
+{
+    struct sol_flow_node *node = data;
+    const char *name;
+
+    name = sol_netctl_service_get_name(service);
+
+    if (name)
+        sol_flow_send_error_packet(node, ENOENT, "Service %s error is %d", name, error);
+}
+
+static struct sol_netctl_service *find_service_by_name(char *service_name)
```

Comment:
It would call API sol_netctl_get_services and sol_netctl_service_get_name in find_service_by_name function , so this function cannot be moved into sol-netctl.h :(